llvm/test/Transforms/NewGVN/pr43441.ll

```diff
@@ -38,7 +42,7 @@ sw.bb:                                            ; preds = %entry
   unreachable
 
 sw.bb19:                                          ; preds = %entry
-  br i1 undef, label %if.then37, label %if.end50
+  br i1 false, label %if.then37, label %if.end50
```

Comment:
why false here?

Reply:
I saw that `%if.then27` is unreachable, so I replaced `undef` with `false`. Couldn't we do that since we already know that it cannot be branched to `%if.then27`? Do we usually leave the unreachable BB handle by the optimizer?

Reply:
Lee, as a matter of policy I think you should just replace undef with fresh function parameters instead of true or false. let's not make the optimizer's job any easier than we have to.

Reply:
Got it! 👌